scene/resources/shape_2d.cpp

```diff
@@ -112,7 +112,7 @@ void Shape2D::_bind_methods() {
 bool Shape2D::is_collision_outline_enabled() {
 #ifdef TOOLS_ENABLED
 	if (Engine::get_singleton()->is_editor_hint()) {
-		return true;
+		return ProjectSettings::get_singleton()->get("debug/shapes/collision/draw_2d_outlines");
```

Comment:
This is the same as line 118, so it makes the whole `#ifdef TOOLS_ENABLED` block redundant.

But I think that's also not a correct change, this setting is meant to configure debug views *at runtime*, not in the editor. See its documentation:

> Sets whether 2D physics will display collision outlines in game when "Visible Collision Shapes" is enabled in the Debug menu.

Reply:
You bring up a good point. I guess the better option would be to make a separate Editor setting for this. I'll get on it.